src/common/execution.py

```diff
@@ -104,8 +105,24 @@ async def get_last_rewards_update() -> RewardVoteInfo | None:
 
 
 async def get_max_fee_per_gas() -> Wei:
-    priority_fee = await execution_client.eth.max_priority_fee  # type: ignore
+    priority_fee = await calculate_median_priority_fee()  # type: ignore
     latest_block = await execution_client.eth.get_block('latest')  # type: ignore
     base_fee = latest_block['baseFeePerGas']
     max_fee_per_gas = priority_fee + 2 * base_fee
     return Wei(max_fee_per_gas)
+
+
+async def calculate_median_priority_fee(block_id: str = 'latest') -> Wei:
+    block = await execution_client.eth.get_block(block_id)
+
+    # collect maxPriorityFeePerGas for all transactions in the block
+    priority_fees = []
+    for tx_hash in block.transactions:
+        tx = await execution_client.eth.get_transaction(tx_hash)
+        if 'maxPriorityFeePerGas' in tx:
+            priority_fees.append(tx.maxPriorityFeePerGas)
+
+    if not priority_fees:
+        return await calculate_median_priority_fee(block['number'] - 1)
```

Comment:
(optional) recursion is not necessary here. The same thing can be done with a loop.